optional_plugins/mail/avocado_result_mail/result_mail.py

```diff
@@ -160,6 +160,10 @@ def _send_email(smtp, sender, rcpt, msg):
 
     @staticmethod
     def _smtp_login_and_send(job, msg):
+        if not job.config.get("plugins.mail.server"):
```

Comment:
Can you explain why only `plugins.mail.server` is checked here?  AFAICT, all of the configuration (server, port, sender, password) will be needed, right?

If that's the case, a check such as:

```python
server, port, sender, password = Mail._get_smtp_config(job)
if not all([server, port, sender, password]):
   ...
```

That suggestion of mine looks a lot `_validate_email_config()` so it seems like it could be refactored a bit and used here too.

Reply:
Hi Cleber, thanks for the feedback i've implemented the validate email config inside of login_and_send to refactor now. Looks like it is all working as intended :)